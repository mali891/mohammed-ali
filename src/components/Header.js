import React from "react"
import { StaticQuery, graphql } from "gatsby"

const TitleAndDescription = ({ data }) => {
  const title = data.site.siteMetadata.title
  const description = data.site.siteMetadata.description

  return (
    <React.Fragment>
      <h1>{title}</h1>
      <h2>{description}</h2>
    </React.Fragment>
  )
}

const Header = () => (
  <StaticQuery
    query={graphql`
      query {
        site {
          siteMetadata {
            title
            description
          }
        }
      }
    `}
    render={data => <TitleAndDescription data={data} />}
  />
)

export default Header
