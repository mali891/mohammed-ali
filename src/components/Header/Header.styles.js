import { css } from "@emotion/core"
import { FONT_SIZES, SPACING } from "../../styles/settings"

export const styles = css`
  height: 100vh;
  width: 100vw;
  margin-bottom: ${SPACING.XL};

  .c-header__container {
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
  }

  .c-header__more-link {
    position: absolute;
    transform: translate(-50%, -50%);
    left: 50%;
    bottom: 2rem;
    font-size: ${FONT_SIZES.XS};
  }
`
