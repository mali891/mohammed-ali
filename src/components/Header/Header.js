import React, { Fragment, useRef } from 'react'
import PropTypes from 'prop-types'
import { StaticQuery, graphql } from 'gatsby'
import { motion, useViewportScroll } from 'framer-motion'

import { Container, HeadingBlock, Link, HeaderScrollTarget } from '../'
import { styles } from './Header.styles'
import { scrollTo } from '../../functions'

const { scrollYProgress } = useViewportScroll()

const renderMainHeader = (title, handleScroll, scrollRef) => (
  <Fragment>
    <header className="c-header" css={styles}>
      <Container className="c-header__container">
        <motion.div
          drag
          dragConstraints={{
            top: -50,
            left: -50,
            right: 50,
            bottom: 50
          }}
        >
          <HeadingBlock type="h1" size="xl" align="center" noMargin>
            {title}
          </HeadingBlock>
        </motion.div>
        <motion.path d="M 0, 20 a 20, 20 0 1,0 40,0 a 20, 20 0 1,0 -40,0" style={{ pathLength: scrollYProgress }} />
      </Container>
      <Link className="c-header__more-link" type="button" onClick={handleScroll}>
        More
      </Link>
    </header>
    <HeaderScrollTarget scrollRef={scrollRef} />
  </Fragment>
)

const Header = React.memo(({ title }) => {
  const scrollRef = useRef(null)
  const handleScroll = () => scrollTo(scrollRef.current)

  return (
    <Fragment>
      {title ? (
        renderMainHeader(title, handleScroll, scrollRef)
      ) : (
        <StaticQuery
          query={graphql`
            query {
              site {
                siteMetadata {
                  title
                  description
                  strapline
                }
              }
            }
          `}
          render={data => renderMainHeader(data.site.siteMetadata.strapline, handleScroll, scrollRef)}
        />
      )}
    </Fragment>
  )
})

Header.propTypes = {
  title: PropTypes.string
}

export default Header
