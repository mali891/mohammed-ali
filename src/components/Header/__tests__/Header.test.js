import React from 'react'
import renderer from 'react-test-renderer'

import Header from '../Header'

describe('Header', () => {
  xit('renders correctly', () => {
    const tree = renderer.create(<Header itle="Default Starter" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
