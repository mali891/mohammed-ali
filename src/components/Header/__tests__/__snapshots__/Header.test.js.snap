// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<DocumentFragment>
  <header
    class="c-header"
    css="[object Object]"
  >
    <div
      class="c-header__container c-container"
      css="[object Object]"
    >
      <div>
        <h1
          class="c-heading c-heading--xl c-heading--no-margin c-heading--align-center"
          css="[object Object]"
        >
          Dummy title
        </h1>
      </div>
      <path
        d="M 0, 20 a 20, 20 0 1,0 40,0 a 20, 20 0 1,0 -40,0"
      />
    </div>
    <button
      class="c-header__more-link c-link u-clr--secondary"
      css="[object Object]"
    >
      More
    </button>
  </header>
  <div
    class="c-header-scroll"
    css="[object Object]"
  />
</DocumentFragment>
`;
